skglm/solvers/base.py

```diff
@@ -0,0 +1,41 @@
+from abc import abstractmethod
+
+
+class BaseSolver():
+    """Base class for solvers."""
+
+    @abstractmethod
+    def solve(self, X, y, datafit, penalty, w_init, Xw_init):
+        """Solve an optimization problem.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            Training data.
+
+        y : array, shape (n_samples,)
+            Target values.
+
+        datafit : instance of Datafit class
+            Datafitting term.
+
+        penalty : instance of Penalty class
+            Penalty used in the model.
+
+        w_init : array, shape (n_features,)
+            Coefficient vector.
+
+        Xw_init : array, shape (n_samples,)
+            Model fit.
```

Comment:
Could we be more precise here? What it stands for and why we require it.

Reply:
makes sense, but the goal of this PR is to move existing code into classes. If we start changing this code, it will be harder to review.